src/NativeScript/GlobalObject.moduleLoader.mm

```diff
@@ -0,0 +1,388 @@
+                                                  CallType callType = JSC::getCallData(function, callData);
+
+                                                  MarkedArgumentBuffer args;
+                                                  args.append(moduleKey);
+                                                  JSValue entry = JSC::call(execState, function, callType, callData, moduleLoader, args);
+                                                  record = jsCast<JSModuleRecord*>(entry.get(execState, Identifier::fromString(execState, "module")));
+
+                                                  return JSValue::encode(jsUndefined());
+                                              }), errorHandler);
+
+                      return JSValue::encode(promise);
+                  }), errorHandler);
+    globalObject->drainMicrotasks();
+
+    if (exception) {
+        execState->vm().throwException(execState, exception);
+        return JSValue::encode(exception);
+    }
+
+    // maybe the require'd module is a CommonJS module?
+    if (JSValue moduleFunction = record->getDirect(execState->vm(), globalObject->_commonJSModuleFunctionIdentifier)) {
+        JSValue module = moduleFunction.get(execState, execState->propertyNames().builtinNames().moduleEvaluationPrivateName());
+        return JSValue::encode(module.get(execState, Identifier::fromString(execState, "exports")));
+    }
+
+    JSModuleRecord::Resolution resolution = record->resolveExport(execState, execState->propertyNames().defaultKeyword);
+    if (resolution.type == JSModuleRecord::Resolution::Type::Resolved) {
+        JSValue defaultExport = record->moduleEnvironment()->get(execState, resolution.localName);
+        ASSERT(!defaultExport.isEmpty());
+        return JSValue::encode(defaultExport);
+    }
+
+    return JSValue::encode(jsUndefined());
+}
+
+static void putValueInScopeAndSymbolTable(VM& vm, JSModuleRecord* moduleRecord, const Identifier& identifier, JSValue value) {
+    JSModuleEnvironment* moduleEnvironment = moduleRecord->moduleEnvironment();
+    SymbolTable* moduleSymbolTable = moduleEnvironment->symbolTable();
+
+    const SymbolTableEntry& entry = moduleSymbolTable->get(identifier.impl());
+    ASSERT(!entry.isNull());
+    moduleEnvironment->variableAt(entry.scopeOffset()).set(vm, moduleEnvironment, value);
+}
+
+JSValue GlobalObject::moduleLoaderEvaluate(JSGlobalObject* globalObject, ExecState* execState, JSValue keyValue, JSValue moduleRecordValue) {
+    JSModuleRecord* moduleRecord = jsDynamicCast<JSModuleRecord*>(moduleRecordValue);
+    if (!moduleRecord) {
+        return jsUndefined();
+    }
+
+    GlobalObject* self = jsCast<GlobalObject*>(globalObject);
+    VM& vm = execState->vm();
+
+    if (JSValue moduleFunction = moduleRecord->getDirect(vm, self->_commonJSModuleFunctionIdentifier)) {
+        NSURL* moduleUrl = [NSURL fileURLWithPath:(NSString*)keyValue.toWTFString(execState).createCFString().get()];
```

Comment:
`createCFString` returns a `WTF::RetainPtr<CFStringRef>` which releases the string as soon as it exits scope.